### DISCUSSION TITLE
Drag to Reoder

### DISCUSSION THREAD

**Discussion by badepaz** [tevelee/SwiftUI-Flow#8] [Ideas]
Hello! I’m sorry if I’m doing this wrong; I’ve been a software engineer for years but this is my first time to try and help an open source project.

But, I really like this framework. Using this has solved a lot of problems for me in my code that I kept trying to solve by fussing with LazyStack.

but anyway, I thought a really cool problem to solve, that isn’t solved even in the latest iOS 17 SDK, would be drag to reorder. Right now you can kind of make it work using the drag and drop delegates on LazyVStack, but it’s a hacky solution. It’d be nice to have something as simple and elegant as the existing List drag to reorder functionality. And making it part of this Flow framework would be awesome, I think.

Any thoughts? I’ve been looking through the source code to see if I could update it to mimic the drag reorder visuals, animation, and simplicity as the UICollectionView methods.

**Comment by tevelee** [comment]
This is a great idea, thanks for starting a discussion @badepaz! I agree it's a problem worth solving for `HFlow`/`VFlow`.

I was looking at https://github.com/globulus/swiftui-reorderable-foreach for inspiration, which uses `onDrag`, `onDrop` and a `DropDelegate`. In fact, they state that "[ReorderableForEach] Works with any SwiftUI layout", so it seems to be already composable with this existing flow layout.
